src/ImageSharp/Formats/Png/PngEncoderCore.cs

```diff
@@ -830,6 +841,42 @@ private void WriteChunk(Stream stream, PngChunkType type, byte[] data, int offse
             stream.Write(this.buffer, 0, 4); // write the crc
         }
 
+        private void ScaleDownFrom8BitArray(ReadOnlySpan<byte> source, Span<byte> result, int bits)
+        {
+            if (bits >= 8)
```

Comment:
Do we also need checks for values like `3,5,7`? But do we really need this check, we cannot enter this method with a higher value?